### ISSUE TITLE
Invoking get_invoices with array input to i_ds results in 400 bad request due to encoding in Xero request in Django

### ISSUE THREAD

**Issue by mattjwbasterfield** [XeroAPI/xero-python#103]
Upon calling accounting_api.get_invoices() with a list of Invoice IDs (i_ds) passed returns a bad request 400 error.

Function called is as follows:
```
@xero_token_required
def get_invoices(self, i_ds=['']):
    xero_tenant_id = get_xero_tenant_id()
    accounting_api = AccountingApi(api_client)
    data = accounting_api.get_invoices(
        xero_tenant_id=xero_tenant_id, i_ds=['e01809d6-c91b-4493-8bf7-96c1f9fea09a','01809d6-c91b-4493-8bf7-96c1f9fea09a'] #i_ds input overridden for testing limited ID list
    )

    response = serialize_model(data)

    return HttpResponse(response)
```

Submitted request to Xero is as follows:
```
GET /api.xro/2.0/Invoices?IDs=e01809d6-c91b-4493-8bf7-96c1f9fea09a%2C01809d6-c91b-4493-8bf7-96c1f9fea09a
```
Xero docs indicate that request pattern should look like:
```
https://api.xero.com/api.xro/2.0/invoices?Statuses=AUTHORISED,PAID&ContactIDs=3138017f-8ddc-420e-a159-e7e1cf9e643d,4b2df4a1-7aa5-4ce3-9e9c-3c55794c5283
```
I am sure the problem is PICNIC but it seems the comma is being encoded to %2C and its resulting in the 400 error from Xero.

Any guidance will be much appreciated.

**Comment by RettBehrens** [comment]
Hi @mattjwbasterfield please confirm what SDK version you're using?

**Comment by mattjwbasterfield** [comment]
@RettBehrens Thanks for the quick reply.

I am quite confident I am going to miss some info you need, so please do point me in the right direction - but:

xero-python version: 1.19.0
python version: 3.8.10
Django version: 4.1.2

**Comment by RettBehrens** [comment]
@mattjwbasterfield nope that's exactly what I needed. Thanks!

**Comment by mattjwbasterfield** [comment]
Great 👍


Kind Regards

Matthew Basterfield

Basterfield

+27 87 095 5357<tel:+27870955357>

________________________________

**Comment by RettBehrens** [comment]
Hmmm. So testing in our sample app, I'm able to successfully call the API for `get_invoices` with an array of two IDs. Here is the code I used:

```
    xero_tenant_id = get_xero_tenant_id()
    accounting_api = AccountingApi(api_client)

    try:
        read_two_invoices = accounting_api.get_invoices(
            xero_tenant_id, i_ds=["20ff01b8-c2d8-49bb-8abf-a9486c9ea665", "3047f972-b6d9-485d-adbc-adcb3a8ebeb4"]
        )
    except AccountingBadRequestException as exception:
        output = "Error: " + exception.reason
        json = jsonify(exception.error_data)
    else:
        output = "Total invoices found:  {}.".format(len(read_two_invoices.invoices))
        json = serialize_model(read_two_invoices)
```

And I can see the request url as 
`http://api-oauth2.xero.com/api.xro/2.0/Invoices?IDs=20ff01b8-c2d8-49bb-8abf-a9486c9ea665%2C3047f972-b6d9-485d-adbc-adcb3a8ebeb4`

So the encoding does not seem to be causing the 400 error.

Can you share your client ID so I can check the logs?

**Comment by RettBehrens** [comment]
Never mind. I found it using the request url. Sure enough, the error response for you request says the supplied parameter IDs cannot be parsed. Found the problem though. You were right when you said it was `PICNIC` 😉 

`20ff01b8-c2d8-49bb-8abf-a9486c9ea665`
`3047f972-b6d9-485d-adbc-adcb3a8ebeb4`

`e01809d6-c91b-4493-8bf7-96c1f9fea09a`
`01809d6-c91b-4493-8bf7-96c1f9fea09a`

One of the two IDs you're providing is short a digit. It also appears you're otherwise submitting the same ID twice?

**Comment by mattjwbasterfield** [comment]
@RettBehrens - most definitely user error!

I was creating an invoice id array from the Xero webhook, then creating/updating the latest invoices. The webhook passes an invoice-id multiple times if there are multiple events on said invoice (understandably). I needed to remove these duplicates from the array.

As an aside, I did need to limit the length of the array and loop else the request to Xero returned a 404 error. I assume because of URL length limits.

Thanks for the immense help - all sorted!
